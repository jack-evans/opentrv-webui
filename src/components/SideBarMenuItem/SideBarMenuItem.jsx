import React, { Component } from 'react'

class SideBarMenuItem extends Component {
  render () {
    return (
      <div>
        SideBarMenuItem
      </div>
    )
  }
}

export default SideBarMenuItem
