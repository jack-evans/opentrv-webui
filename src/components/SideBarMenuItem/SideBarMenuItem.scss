
.SideBarMenuItem {
  align-items: center;
  display: flex;
  height: 50px;
  justify-content: flex-start;
  padding-left: 40px;
  padding-right: 10px;
  width: 250px;

  .SideBarMenuItem__content {
    color: #000;
    text-align: left;
    text-decoration: none;
  }
}