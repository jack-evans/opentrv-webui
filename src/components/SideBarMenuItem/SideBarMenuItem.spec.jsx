import React from 'react'
import { shallow } from 'enzyme'
import SideBarMenuItem from './SideBarMenuItem'

describe('SideBarMenuItem.jsx', () => {
  it('renders without crashing', () => {
    const expectedRender = (
      <div className='SideBarMenuItem'>
        <a className='SideBarMenuItem__content' href='https://test.com' target='_blank' rel='noopener'>
          <h3>this is a test</h3>
        </a>
      </div>
    )
    expect(shallow(<SideBarMenuItem href='https://test.com' text='this is a test' />).contains(expectedRender)).toBe(true)
  })
})
