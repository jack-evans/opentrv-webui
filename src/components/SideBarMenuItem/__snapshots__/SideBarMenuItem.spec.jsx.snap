// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBarMenuItem.jsx renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SideBarMenuItem
    href="https://test.com"
    text="this is a test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <span
        className="SideBarMenuItem__content"
      >
        <h3>
          this is a test
        </h3>
      </span>,
      "className": "SideBarMenuItem",
      "href": "https://test.com",
      "rel": "noopener",
      "target": "_blank",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h3>
          this is a test
        </h3>,
        "className": "SideBarMenuItem__content",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "this is a test",
        },
        "ref": null,
        "rendered": "this is a test",
        "type": "h3",
      },
      "type": "span",
    },
    "type": "a",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span
          className="SideBarMenuItem__content"
        >
          <h3>
            this is a test
          </h3>
        </span>,
        "className": "SideBarMenuItem",
        "href": "https://test.com",
        "rel": "noopener",
        "target": "_blank",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h3>
            this is a test
          </h3>,
          "className": "SideBarMenuItem__content",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "this is a test",
          },
          "ref": null,
          "rendered": "this is a test",
          "type": "h3",
        },
        "type": "span",
      },
      "type": "a",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
